### ISSUE TITLE
Segment-Anything to remove noise

### ISSUE THREAD

**Issue by Ashoka74** [playcanvas/super-splat#6]
I am not familiar with 3D splats and the feasibility of this proposition but would it be possible to use segment-anything to recognise the main object on 2D projection with a click and remove all noise outside coordinates of the contour area?

**Comment by slimbuck** [comment]
Yeah it would be great to have something like this.

We did a quick test with an open source implementation of dbscan (a clustering algorithm) which looked promising, but ran extremely slowly.

We'll be trying a few things when we have time, but feel free to have a go yourself too. This is probably the functionality we need most.

**Comment by TonyDua** [comment]
> Yeah it would be great to have something like this.
> 
> We did a quick test with an open source implementation of dbscan (a clustering algorithm) which looked promising, but ran extremely slowly.
> 
> We'll be trying a few things when we have time, but feel free to have a go yourself too. This is probably the functionality we need most.


Yes! I really need clustering algorithms like DBSCAN to remove outliers in 3DGS. Manual cleaning is very time-consuming and meticulous. I once implemented a DBSCAN-like clustering algorithm in Metashape to remove outliers, but encountered the problem of slow computation. I tried adding KDTree, but the speed was still slow. Later, I tried a filtering function where I selected a reference point in the viewport, specified a distance, created a box with sides of that distance, selected it in the viewport, and then ran DBSCAN. This approach was much faster. External noise points can be directly removed using features like rectangular selection boxes, but it's difficult to quickly select outliers on the main surface. By only computing points within the box I set (i.e., the main body range), I solved the problem of slow speed. This script runs very well, and I also added a feature to filter points by color.

Unfortunately, due to the algorithmic characteristics of 3DGS, it only requires a sparse point cloud for Gaussian model initialization. Even if I clean the alignment points in Metashape exceptionally well, 3DGS training will still learn these points back, which is its robustness. It's really frustrating. So, I started using your super-splat tool to clean up 3DGS, but it would be great if you could incorporate DBSCAN to quickly select outliers!

Oh, thank you again for developing super-splat. It's truly an amazing tool! If you have any ideas about using DBSCAN to clean outliers, I would be very happy to discuss with you! Thanks again!

**Comment by slimbuck** [comment]
Ideally floaters would be removed at the training stage so we don't have deal with this at all.

BTW have you tried using the new "rings" mode for removing floaters? I find it's easier and faster than before.

**Comment by RobLinkA** [comment]
I've tried with [SAMURAI](https://github.com/yangchris11/samurai) and it really worked!
![Original](https://github.com/user-attachments/assets/f4413bec-130f-47cf-af6c-c8acbada4344)
![Masked](https://github.com/user-attachments/assets/9e7af316-ad70-4244-baf5-8f3746ce6cc6)
![Training](https://github.com/user-attachments/assets/ba12a657-b799-4027-b748-4ef5743f748d)
You can check the result here: https://3d.explorerglobal.cn/model/6l7vdmlj

**Comment by willeastcott** [comment]
Closing as superseded by #512, the central issue tracking AI-assisted 3DGS scene segmentation.
